sdk/nodejs/tests/automation/localWorkspace.spec.ts

```diff
@@ -924,6 +924,25 @@ describe("LocalWorkspace", () => {
             await stack.workspace.removeStack(stackName);
         }
     });
+    it(`runs an inline program that exits gracefully`, async () => {
+        const program = async () => {
+            Promise.resolve();
```

Comment:
It's really strange that I have to put this here or it hangs.

Reply:
Strange, I'm not seeing it hang locally without this. Trying it out here: https://github.com/pulumi/pulumi/pull/14465

Reply:
Yeh this is odd, resolve is just going to create a promise resolved to undefined? If this has to stay in it needs a really good comment to go with it, because it looks like dead code.